modules/aruco/test/test_charucodetection.cpp

```diff
@@ -235,91 +240,101 @@ CV_CharucoPoseEstimation::CV_CharucoPoseEstimation() {}
+                        bottomLeft = getMarkerById(board->ids[2], corners, ids);
+                    } else {
+                        offset = Point2f(aruco_offset, aruco_offset);
+                        topLeft = getMarkerById(board->ids[0], corners, ids);
+                        bottomLeft = getMarkerById(board->ids[2], corners, ids);
+                    }
+                    vector<Point2f> axes = getAxis(cameraMatrix, distCoeffs, rvec, tvec, board->getSquareLength(), offset);
+                    ASSERT_NEAR(topLeft[0].x, axes[1].x, 3.f);
+                    ASSERT_NEAR(topLeft[0].y, axes[1].y, 3.f);
+                    ASSERT_NEAR(bottomLeft[0].x, axes[2].x, 3.f);
+                    ASSERT_NEAR(bottomLeft[0].y, axes[2].y, 3.f);
 
-                // check estimate result
-                vector< Point2f > projectedCharucoCorners;
+                    // check estimate result
+                    vector< Point2f > projectedCharucoCorners;
 
-                projectPoints(board->chessboardCorners, rvec, tvec, cameraMatrix, distCoeffs,
-                              projectedCharucoCorners);
+                    projectPoints(board->chessboardCorners, rvec, tvec, cameraMatrix, distCoeffs,
+                                projectedCharucoCorners);
 
-                for(unsigned int i = 0; i < charucoIds.size(); i++) {
+                    for(unsigned int i = 0; i < charucoIds.size(); i++) {
 
-                    int currentId = charucoIds[i];
+                        int currentId = charucoIds[i];
 
-                    if(currentId >= (int)board->chessboardCorners.size()) {
-                        ts->printf(cvtest::TS::LOG, "Invalid Charuco corner id");
-                        ts->set_failed_test_info(cvtest::TS::FAIL_MISMATCH);
-                        return;
-                    }
+                        if(currentId >= (int)board->chessboardCorners.size()) {
+                            ts->printf(cvtest::TS::LOG, "Invalid Charuco corner id");
+                            ts->set_failed_test_info(cvtest::TS::FAIL_MISMATCH);
+                            return;
+                        }
 
-                    double repError = cv::norm(charucoCorners[i] - projectedCharucoCorners[currentId]);  // TODO cvtest
+                        double repError = cv::norm(charucoCorners[i] - projectedCharucoCorners[currentId]);  // TODO cvtest
 
 
-                    if(repError > 5.) {
-                        ts->printf(cvtest::TS::LOG, "Charuco corner reprojection error too high");
-                        ts->set_failed_test_info(cvtest::TS::FAIL_MISMATCH);
-                        return;
+                        if(repError > 5.) {
+                            ts->printf(cvtest::TS::LOG, "Charuco corner reprojection error too high");
+                            ts->set_failed_test_info(cvtest::TS::FAIL_MISMATCH);
+                            return;
+                        }
```

Comment:
I propose to use `ASSERT_LE() << "Charuco corner reprojection error too high"`. Instead of condition block.

Reply:
done.